crates/js_api/src/gui/order_operations.rs

```diff
@@ -0,0 +1,293 @@
+            .cloned()
+    }
+
+    fn get_all_vaults(&self) -> Vec<OrderIO> {
+        let mut all_vaults: Vec<_> = self.deployment.deployment.order.inputs.clone();
+        let filtered_outputs: Vec<OrderIO> = self
+            .deployment
+            .deployment
+            .order
+            .outputs
+            .iter()
+            .filter(|output| !all_vaults.contains(*output))
+            .cloned()
+            .collect();
+        all_vaults.extend(filtered_outputs);
+        all_vaults
+    }
+
+    fn get_deposits_as_map(&self) -> HashMap<Address, String> {
+        self.get_deposits()
+            .into_iter()
+            .map(|d| (d.address, d.amount))
+            .collect()
+    }
+
+    async fn get_token_decimals(&self, rpc_url: &str, token: Address) -> Result<u8, GuiError> {
+        let client = ReadableClientHttp::new_from_url(rpc_url.to_string())?;
+        let decimals = client
+            .read(ReadContractParameters {
+                address: token,
+                call: decimalsCall {},
+                block_number: None,
+            })
+            .await?
+            ._0;
+        Ok(decimals)
+    }
+
+    async fn get_deposit_args(
+        &self,
+        rpc_url: &str,
+        deposits_map: &HashMap<Address, String>,
+        order_io: &OrderIO,
+    ) -> Result<DepositArgs, GuiError> {
+        let decimals = if let Some(decimals) = order_io.token.decimals {
+            decimals
+        } else {
+            self.get_token_decimals(rpc_url, order_io.token.address)
+                .await?
+        };
+
+        Ok(DepositArgs {
+            token: order_io.token.address,
+            vault_id: order_io.vault_id.ok_or(GuiError::VaultIdNotFound)?,
+            amount: parse_units(
+                &deposits_map
+                    .get(&order_io.token.address)
+                    .ok_or(GuiError::TokenNotFound)?,
+                decimals,
+            )?
+            .into(),
+        })
+    }
+
+    async fn check_allowance(
+        &self,
+        orderbook: &Orderbook,
+        deposit_args: &DepositArgs,
+        owner: &str,
+    ) -> Result<TokenAllowance, GuiError> {
+        let allowance = deposit_args
+            .read_allowance(
+                Address::from_str(owner)?,
+                TransactionArgs {
+                    orderbook_address: orderbook.address,
+                    rpc_url: orderbook.network.rpc.to_string(),
+                    ..Default::default()
+                },
+            )
+            .await?;
+        Ok(TokenAllowance {
+            token: deposit_args.token,
+            allowance,
+        })
+    }
+
+    /// Check allowances for all inputs and outputs of the order
+    ///
+    /// Returns a vector of [`TokenAllowance`] objects
+    #[wasm_bindgen(js_name = "checkAllowances")]
+    pub async fn check_allowances(&self, owner: String) -> Result<JsValue, GuiError> {
+        let all_vaults = self.get_all_vaults();
```

Comment:
instead of checking allowances for all vaults, shouldn't we be checking allowances for all deposits?

we don't need to check allowance for input vault tokens (that's for tokens coming in, so you don't deposit into an input vault). we also don't need to check allowance for output vault tokens that aren't being deposited into.

Reply:
ok i wasn't sure before implementing this but i get the picture now, will do it based on this